src/wagtail_localize_smartling/models.py

```diff
@@ -365,3 +370,124 @@ def get_or_create_from_source_and_translation_data(
         # If we get here we've got a proper background worker, so we can safely
         # enqueue the syncing of the job.
         background.enqueue(sync_job, args=(job.pk,), kwargs={})
+
+
+class LandedTranslationTaskManager(models.Manager):
+    def incomplete(self):
+        return self.filter(
+            completed_on__isnull=True,
+            cancelled_on__isnull=True,
+        )
+
+    def create_from_source_and_translation(
+        self,
+        source_object: models.Model,
+        translated_locale: Locale,
+    ) -> "LandedTranslationTask":
+        """
+        Make a LandedTranslationTask for all users of the translation-approval
+        group, for the relevant translation of the source object.
+
+        Note that the source object is the instance that was translated (from
+        the Job), not the resulting translation. This is why we need to look the
+        latter up via the relevant locale.
+
+        We do store the resulting translated object (e.g. Page or Snippet) as
+        the target of the generic FK.
+        """
+
+        translated_object = source_object.get_translations().get(  # pyright: ignore[reportAttributeAccessIssue]
+            locale=translated_locale
+        )
+
+        c_type = ContentType.objects.get_for_model(translated_object)
+
+        task, created = LandedTranslationTask.objects.get_or_create(
+            content_type=c_type,
+            object_id=translated_object.pk,
+            relevant_locale=translated_locale,
+            completed_on__isnull=True,
+            cancelled_on__isnull=True,
+        )
+        action = "made" if created else "found"
+
+        msg = (
+            f"Translation-approval task {action} for {c_type.name}#{translated_object.pk}"
+            f" in {translated_locale.language_name}."
+        )
+        logger.info(msg)
+
+        return task
+
+
+class LandedTranslationTask(models.Model):
+    """
+    A custom task prompting members of a particular Group to review and
+    publish a particular Page or Snippet, which has just had translations land.
+
+    Note that this is _not_ a subclass of Task, and we don't want it to sit
+    within a workflow because Workflows a) don't support Snippets and b)
```

Comment:
note that snippets can support workflows - https://docs.wagtail.org/en/stable/topics/snippets/features.html#enabling-workflows-for-snippets so this is not entirely true :)

but the other point stands strong